pkg/opendsme/include/opendsme/opendsme.h

```diff
@@ -0,0 +1,284 @@
+ * GTS transmissions. A value equal or higher than 3 allows transmission of 127 bytes
+ * frames.
+ * Increasing the number of superframes per multisuperframe increases
+ * on one hand transmission delay, because the period of a single GTS is the duration of the
+ * multisuperframe. On the other hand it decreases energy consumption, because a
+ * device transmit less often.
+ * Increasing the beacon interval extends the number of coordinators in wireless
+ * reach and reduces energy consumption. However, it slows the scanning and joining
+ * procedure.
+ *
+ * The configuration of SO, MO and BO must be compliant with:
+ * ```
+ * 0 <= SO <= MO <= BO < 15
+ * ```
+ *
+ * ## CAP Reduction
+ *
+ * With the purpose of reducing energy consumption and extending the GTS
+ * resources even more, the MAC defines a CAP Reduction mode that turns
+ * all CAP (except the first one in a multisuperframe) into CFP. This adds
+ * 8 GTS per superframe that reduces CAP.
+ *
+ * With CAP Reduction on, increasing the number of superframes per multisuperframe
+ * improves energy efficiency, because nodes spend less time on CAP. However,
+ * note that this renders less CAP time which is required for slot negotiation.
+ *
+ * ## GTS coordinates
+ *
+ * A slot is defined by a superframe ID, a superframe slot and
+ * a channel offset. The superframe ID is contained in `{0..n-1}`, where
+ * `n` is the number of superframes per multisuperframe. The ID of the first superframe
+ * (the one aligned with the start of a multisuperframe) is 0.
+ *
+ * The superframe slot values depend on the superframe ID.
+ * ```
+ * sf_slot = {0..6} if superframe_ID==0
+ * sf_slot = {8..14} if superframe_ID!=0 && cap_reduction=0
+ * sf_slot = {0..14} if superframe_ID!=0 && cap_reduction=1
+ * ```
+ *
+ * The channel offset is `{0..m-1}`, with `m` the number of channels.
+ *
+ * # Usage in RIOT
+ *
+ * The RIOT implementation uses the `DSMEAdaptionLayer` wrapper internally (provided by
+ * openDSME) to manage scanning/joining and automatic slot allocation. The MAC
+ * implementation defines several scheduling strategies, but the default is
+ * used (Traffic-aware and Predictive Scheduling).
+ * The port introduces an openDSME GNRC Network Interface, that can be used via
+ * @ref net_gnrc_netapi and @ref net_gnrc_netreg.
+ *
+ * The following network options (@ref net_netopt) are used for DSME setup.
+ *
+ * Configuration:
+ * - @ref NETOPT_PAN_COORD: Set PAN coordinator or Child role
+ * - @ref NETOPT_GTS_ALLOC: Allocate a GTS slot manually (only available via
+ *   @ref CONFIG_IEEE802154_DSME_STATIC_GTS).
+ * - @ref NETOPT_GTS_TX: Set the next transmission to either GTS or CSMA-CA.
+ * - @ref NETOPT_ACK_REQ: Set the ACK Request bit to enable confirmed
+ *   transmissions.
+ * - @ref NETOPT_LINK: Start the DSME network (PAN Coordinator) or join an
+ *   existing network (Child).
+ *
+ * ## MAC configuration
+ *
+ */
+#ifndef PKG_OPENDSME_H
+#define PKG_OPENDSME_H
```

Comment:
```suggestion
#ifndef OPENDSME_OPENDSME_H
#define OPENDSME_OPENDSME_H
```

courtesy of headerguards/check.h